### PR TITLE
Add in-app restore + retention GC for deleted trips/costs (Vibe Kanban)

### LINKED ISSUE #7: Backup restore and garbage collection system
## Description
With the delete functionality creating backups of deleted trips and cost trackers, we need a system to manage these backups effectively. This includes both restoration capabilities and periodic cleanup to prevent unlimited storage growth.

### Restore System
- [ ] UI to view deleted trips/cost trackers from backups
- [ ] Restore functionality to bring back deleted items
- [ ] Conflict handling if trip ID already exists
- [ ] Restore validation and data integrity checks

### Garbage Collection
- [ ] Configurable backup retention period (e.g., 30/90 days)
- [ ] Scheduled cleanup of old backups
- [ ] Manual cleanup interface for admins
- [ ] Backup compression for older items
- [ ] Backup metadata tracking (deletion date, reason, etc.)

### Admin Interface
- [ ] Backup management dashboard
- [ ] Bulk operations (restore multiple items, bulk delete backups)
- [ ] Backup storage size monitoring
- [ ] Export/import backup functionality

## Technical Considerations
- Backup directory structure and organization
- Database/index for efficient backup queries
- Background job system for scheduled cleanup
- Backup integrity verification
- Storage optimization strategies

## Dependencies
- Requires delete functionality (#5) to be implemented first
- Should coordinate with legacy JSON retirement (#6)

### PR DESCRIPTION
**Pull request by bdamokos**
Fixes #7

## What changed
- **Cost tracker deletion now creates a backup**: added `DELETE /api/cost-tracking?id=...` and a delete affordance in the Cost Tracking admin list. This removes `costData` from the trip and clears any `costTrackingLinks` on locations/routes/accommodations so the trip doesn't keep dangling references.
- **In-app backup management + restore**: added server routes to list backups, verify integrity, restore, and delete backup files.
- **Backup retention / garbage collection**: added a GC endpoint that deletes backups older than a retention window while keeping at least the newest N backups.
- **Admin UI**: added an Admin “Backups” tab for browsing/searching backups, restoring (with conflict handling), deleting backups, and running GC (dry-run supported).
- **Test coverage**: added unit tests for trip restore, cost restore (including link re-hydration), and GC behavior.

## Why
Accidental deletions of trips (or cost trackers within a trip) currently require manual file recovery. This PR adds a first-class safety net: automatic backups on delete, a self-serve restore flow, and retention controls so `data/backups` doesn't grow without bounds.

## Implementation details
- **Backup formats / storage**
  - Backups are stored as JSON files under `data/backups/` and indexed in `data/backup-metadata.json`.
  - Trip deletions create `deleted-trip-<tripId>-<timestamp>.json`. Cost deletions create `deleted-cost-<tripId>-<timestamp>.json`.
  - The metadata layer tracks checksums for integrity verification and supports directory→metadata resync.
- **Restore semantics**
  - **Trip restore**: recreates the trip file from a selected backup (optionally to a new ID or overwriting an existing trip).
  - **Cost restore**: reattaches `costData` to an existing trip and merges `costTrackingLinks` back onto matching travel entities by ID (so travel edits since deletion are preserved).
- **Safety / auth**
  - Backup APIs are admin-domain gated.
  - GC can also be called with `BACKUP_GC_TOKEN` via `x-backup-gc-token` (intended for cron/automation).
  - Backup deletion validates paths to ensure only files within the backups directory are removed.
- **Retention policy**
  - `POST /api/backups/gc` uses `retentionDays` and `keepLatest` (defaults: 30 days / keep 20), with a `dryRun` option. Env overrides: `BACKUP_RETENTION_DAYS`, `BACKUP_GC_KEEP_LATEST`.

## Testing
- `bun run test:unit`
- `bun run build`

This PR was written using [Vibe Kanban](https://vibekanban.com)

## Summary by CodeRabbit

* **New Features**
  * Admin Backups Manager UI and Backups tab for viewing, searching, restoring, and deleting backups; configurable garbage-collection with dry-run.
  * New server endpoints to list/synchronize backups, restore/delete individual backups, and run GC.
  * Per-item delete confirmation UI for cost trackers.

* **Data / Recovery**
  * Deletions now create recoverable backups for trips and cost trackers; restoration supports conflict handling and safe-ID options.

* **Tests**
  * New unit and integration tests covering backup, restore, deletion, scrubbing, and GC flows.

* **Documentation**
  * README updated with backup behavior, storage location, retention defaults, and GC token.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by bdamokos** [comment]
Follow-up on review notes (non-inline comments):

- @coderabbitai modal a11y nit: the cost-tracker delete confirmation is now a proper dialog (`role="dialog"`, `aria-modal`, `aria-labelledby`/`aria-describedby`) with focus management/trap (see `src/app/admin/components/CostTracking/CostTrackerList.tsx`).
- @greptile-apps retention/config nit: documented defaults + env overrides in `README.md` (`BACKUP_RETENTION_DAYS`, `BACKUP_GC_KEEP_LATEST`, `BACKUP_GC_TOKEN`).
- @greptile-apps restore ID nit: restore ID generation uses a longer random suffix (`slice(2, 14)`) for better collision resistance (see `src/app/admin/components/Backups/BackupsManager.tsx`).
- Type-safety nit: `mergeCostTrackingLinks` is typed as `CostTrackingLink[]` instead of `unknown` (see `src/app/lib/unifiedDataService.ts`).

One note: the route handlers here still use `params: Promise<...>` (consistent with other API routes in this repo).

(If you still want it, I can follow up on the edge case where `clearCostTrackingLinks` returns early when `travelData` is missing, which means accommodations links may not be scrubbed in that scenario.)

File: src/app/api/backups/[backupId]/route.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+export async function GET(_request: NextRequest, context: { params: Promise<{ backupId: string }> }) {
+  try {
+    const isAdmin = await isAdminDomain();
+    if (!isAdmin) {
+      return NextResponse.json({ error: 'Unauthorized' }, { status: 403 });
+    }
+
+    const { backupId } = await context.params;
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/api/backups/[backupId]/route.ts]: Thanks — tried the plain-object context.params typing, but Next.js type-check in this repo rejects it for route handlers (build fails). Keeping the Promise-based params shape here to satisfy the project’s App Router typing (see commit 2952ff0).

File: src/app/admin/components/Backups/BackupsManager.tsx [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+function formatBytes(bytes: number): string {
+  if (!Number.isFinite(bytes) || bytes <= 0) return '0 B';
+  const units = ['B', 'KB', 'MB', 'GB'];
+  const idx = Math.min(units.length - 1, Math.floor(Math.log(bytes) / Math.log(1024)));
+  const value = bytes / Math.pow(1024, idx);
+  return `${value.toFixed(value >= 10 || idx === 0 ? 0 : 1)} ${units[idx]}`;
+}
+
+function generateSafeTripId(): string {
+  return `restored${Date.now().toString(36)}${Math.random().toString(36).substring(2, 8)}`;
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/admin/components/Backups/BackupsManager.tsx]: Adjusted: restore-copy IDs now use a longer random suffix (slice(2, 14)) to reduce collision risk (see BackupsManager update).

File: src/app/api/backups/[backupId]/route.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          return NextResponse.json({ error: 'Trip already has cost tracking data', conflict: true }, { status: 409 });
+        }
+        if (message.includes('Trip') && message.includes('not found')) {
+          return NextResponse.json({ error: 'Trip not found' }, { status: 404 });
+        }
+        if (message.includes('Backup') && message.includes('not found')) {
+          return NextResponse.json({ error: 'Backup not found' }, { status: 404 });
+        }
+        throw error;
+      }
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/api/backups/[backupId]/route.ts]: Agreed — replaced string-matching with typed errors (ConflictError/NotFoundError/ValidationError) thrown from the service layer and handled via instanceof in the route handler (see src/app/lib/errors.ts + backups route).

File: src/app/admin/components/CostTracking/CostTrackerList.tsx [hunk 4]
```diff
@@ -10,13 +11,141 @@ interface CostTrackerListProps {
...
+              >
+                {deleteDialog.isDeleting && (
+                  <div className="w-4 h-4 border-2 border-white border-t-transparent rounded-full animate-spin"></div>
+                )}
+                {deleteDialog.isDeleting ? 'Deleting...' : 'Delete'}
+              </button>
+            </div>
+          </div>
+        </div>
+      )}
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/admin/components/CostTracking/CostTrackerList.tsx]: Good catch on the edge case — fixed  to always scrub  even when  is missing, and added a unit test for the cost-only trip scenario (commit 71da0e7).

File: src/app/lib/backupService.ts [hunk 5]
```diff
@@ -333,6 +339,81 @@ export class BackupService {
...
+  /**
+   * Garbage-collects old backups based on retention policy, keeping at least `keepLatest` most recent backups.
+   */
+  async garbageCollect(options?: {
+    retentionDays?: number;
+    keepLatest?: number;
+    dryRun?: boolean;
+  }): Promise<{ deleted: BackupMetadata[]; kept: BackupMetadata[]; errors: string[] }> {
+    const retentionDays = Math.max(0, options?.retentionDays ?? 30);
+    const keepLatest = Math.max(0, options?.keepLatest ?? 20);
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/lib/backupService.ts]: Done: retention defaults (30 days / keep 20) and env vars (BACKUP_RETENTION_DAYS, BACKUP_GC_KEEP_LATEST, BACKUP_GC_TOKEN) are documented in README.md.

File: src/app/admin/components/CostTracking/CostTrackerList.tsx [hunk 6]
```diff
@@ -10,13 +11,201 @@ interface CostTrackerListProps {
...
+      if (event.shiftKey) {
+        if (!active || active === first || !container.contains(active)) {
+          event.preventDefault();
+          last.focus();
+        }
+      } else {
+        if (active === last) {
+          event.preventDefault();
+          first.focus();
+        }
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/admin/components/CostTracking/CostTrackerList.tsx]: Implemented the remaining focus-trap edge case: forward-Tab now wraps back into the dialog when focus is outside the container (commit c20a99a).

File: src/app/lib/unifiedDataService.ts [hunk 7]
```diff
@@ -31,7 +34,177 @@ async function ensureBackupDir() {
...
+    ...current,
+    travelData: {
+      ...current.travelData,
+      locations: mergeLocations(current.travelData.locations, restored.travelData.locations),
+      routes: mergeRoutes(current.travelData.routes, restored.travelData.routes),
+      days: mergeDays(current.travelData.days as JourneyPeriod[] | undefined, restored.travelData.days as JourneyPeriod[] | undefined)
+    },
+    accommodations: mergeAccommodations(current.accommodations, restored.accommodations)
+  };
+}
```
[the remark this answers is not in the document]
Comment (bdamokos) [reply src/app/lib/unifiedDataService.ts]: Fixed: mergeRestoredLinksIntoTrip now merges accommodations even when travelData is missing; added a regression unit test for cost restore on travelData-missing trips (commit c20a99a).

**Comment by bdamokos** [comment]
Follow-up: addressed the latest CodeRabbit edge cases (label associations in BackupsManager; forward-Tab focus trap in the cost delete modal; restore merge now includes accommodations even when travelData is missing). Changes are in commit c20a99a; tests/build run locally.
